gradio/blocks.py

```diff
@@ -978,10 +978,47 @@ def deserialize_data(self, fn_index: int, outputs: List[Any]) -> List[Any]:
 
         return predictions
 
+    def validate_inputs(self, fn_index: int, inputs: List[Any]):
+        block_fn = self.fns[fn_index]
+        dependency = self.dependencies[fn_index]
+
+        dep_inputs = dependency["inputs"]
+
+        # This handles incorrect inputs when args are changed by a JS function
+        # Only check not enough args case, ignore extra arguments (for now)
+        # TODO: make this stricter?
+        if len(inputs) < len(dep_inputs):
+            name = f" ({block_fn.name})" if block_fn.name and block_fn.name != "<lambda>" else ""
+
+            wanted_args = []
+            received_args = []
+            for input_id in dep_inputs:
+                block = self.blocks[input_id]
+                wanted_args.append(str(block))
+            for inp in inputs:
+                if isinstance(inp, str):
+                    v = f'"{inp}"'
+                else:
+                    v = str(inp)
+                received_args.append(v)
+
+            wanted = ", ".join(wanted_args)
+            received = ", ".join(received_args)
+
+            # JS func didn't pass enough arguments
+            raise ValueError(f"""An event handler{name} didn't receive enough input values (needed: {len(dep_inputs)}, got: {len(inputs)}).
```

Comment:
Can you add a test for this case (js function not passing enough args)? Thank you!

Reply:
Can we run JS in tests? I'm fine with mocking it otherwise

Reply:
Actually it's probably fine to leave as is in this case!